connector/connect/common/src/main/scala/org/apache/spark/sql/connect/common/DataTypeProtoConverter.scala

```diff
@@ -270,6 +271,8 @@ object DataTypeProtoConverter {
               .build())
           .build()
 
+      case VariantType => ProtoDataTypes.VariantType
```

Comment:
This should also have a conversion in the other direction.

Reply:
I might be missing something. Is the conversion above---`case proto.DataType.KindCase.VARIANT => VariantType`---the conversion in the other direction?

Reply:
Yeah the conversion to the other side.

Connect can also pass a schema to the server side.